### ISSUE TITLE
Implement key generating algorithm

### ISSUE THREAD

**Issue by Wynadorn** [d0k3/GodMode9#786]
**Describe the feature you'd like**
GodMode9 could be changed to no longer have to rely on `encTitleKeys.bin` and `decTitleKeys.bin`. These keys can be generated as long as the user provides a gm9 with a 32-bit common key and the correct passphrase.

**For further info please see**
* [3DS/Wii U titlekey generation algorithm leaked](https://gbatemp.net/threads/3ds-wii-u-titlekey-generation-algorithm-leaked.566318/#post-9075393)
* [[Release] Kii U Generator](https://gbatemp.net/threads/release-kii-u-generator.566930/) code available here -> [dojafoja/Kii-U-Generator](https://github.com/dojafoja/Kii-U-Generator)
* [My findings, using the Kii U Generator](https://gbatemp.net/threads/my-findings-using-the-kii-u-generator.567844/)

**Comment by d0k3** [comment]
You're correct in that GM9 could be changed, but that algorithm you're referring to was leaked and not reverse-engineered, so it's a bit questionable. Also, you're aware that the passphrase is the same only for *most* but not all games?

Anyways, this has been discussed in the past, and while I'd really, really like to have this algorithm in GM9, I'd rather not. I'd be okay with this algorithm being added via some external (ie. not downloadable from here or included in the GM9 source code) means, but that would be somewhat similar to just having a complete encTitlekeys database, I guess..

I'm keeping this open just in case any good ideas will be coming in.

**Comment by urherenow** [comment]
In similar cases, the solution may have been too messy for your tastes. It would involve having to provide the extra code from an external source (say... pastebin or something), and in your code, add a build rule to include the file(s) (plural if both an .h and a .c is required) like make keygen=1, and appropriate ifdefs within the appropriate menus to add the option.
